ogre2/src/Ogre2GpuRays.cc

```diff
@@ -295,7 +295,7 @@ void Ogre2LaserRetroMaterialSwitcher::passPreExecute(
   const Ogre::HlmsBlendblock *noBlend =
     hlmsManager->getBlendblock(Ogre::HlmsBlendblock());
 
-  const std::string laserRetroKey = "laser_retro";
+  static const std::string laserRetroKey = "laser_retro";
```

Comment:
`std::get(variant)` can throw exception if variant do not holds the alternative T it throws [std::bad_variant_access](https://en.cppreference.com/w/cpp/utility/variant/bad_variant_access).
Also seems modern compilers can detect `hold_alternatives->get` pattern and generate quite straightforward code(assembly sample in a godbolt, link above)

Technically `get_if` is slightly safer, due to it preventing the case when developer can do something like this:
`if (std::hodls_alternative<A>(v))  result = std::get<B>(v);`, but I agree, it will be much longer

Idea behind a `static` is to construct object once, not on each function call. But perhaps compiler can do optimizations. I'll remove it